sdk/python/kfp/components/_editing.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2020 The Kubeflow Pipelines authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import warnings
+from .structures import ComponentReference
+from ._component_store import ComponentStore
+
+
+def edit_component_yaml(component_ref: ComponentReference):
+    '''Generates code to edit the component YAML text.
+
+    This function can work with components loaded from URLs, files or text.
+    This function is used indirectly by calling `op.edit()` method.
+    If used in a notebook, adds cell with the component editing code.
+    Otherwise returns the code as string.
+
+    Example::
+
+        my_op = components.load_component_from_*(...)
+        my_op.edit()
+    '''
+    editing_code = generate_code_to_edit_component_yaml(component_ref=component_ref)
+    # If we're in a notebook, add a new cell. Otherwise return the editing code as string
+    try:
+        _ipython_create_new_cell(editing_code)
+    except:
+        return editing_code
+
+
+def generate_code_to_edit_component_yaml(
+    component_ref: ComponentReference, component_store: ComponentStore = None,
+):
+    '''Generates code to edit the component YAML text.
+
+    This function can work with components loaded from URLs, files or text.
+    '''
+    component_spec = component_ref.spec
+    if not component_spec:
+        if not component_store:
+            component_store = ComponentStore.default_store
+        component_ref = component_store._load_component_spec_in_component_ref(component_ref)
+        component_spec = component_ref.spec
+
+    component_data = component_spec._data
+    local_path = getattr(component_spec, '_local_path', None)
+    if not local_path:
+        if component_ref.url:
```

Comment:
I think we should at least add some logging message here for this implicit side effect, to inform users that compiler is writing out to a local file.